### PR TITLE
Jetpack Search: enable on all WP.com sites

### PR DESCRIPTION
**Pull request by gibrown**
#### Changes proposed in this Pull Request

* Allow purchases on private wp.com sites (simple and atomic)
* Clean up and simplify the purchase code a bit.
* Allow canceling JPS on Atomic sites without contacting support
* Rewrite and update the settings code and upgrade nudge

#### Testing instructions

Types of sites to test against:
* Jetpack public site
* WP.com public simple site
* WP.com private simple site (new)
* WP.com public atomic site
* WP.com private atomic sites (new)

Flows to test:
* Purchase through /purchase-product/jetpack_search/monthly (should now work for wp.com private sites)
* Purchase through /checkout/\<site\>/jetpack_search (should now work for wp.com private sites)
* Refund through /purchases/subscriptions/ (for atomic sites it should no longer require contacting support)
* /settings/performance/ should nudge the user to purchase if they don't have a JPS subscription. Otherwise it shows settings and link to Customizer

Screen Shot 2020-10-06 at 6 39 21 PM

Screen Shot 2020-10-06 at 6 39 33 PM

Screen Shot 2020-10-06 at 6 39 40 PM

Screen Shot 2020-10-06 at 6 39 48 PM

### CODE REVIEW

**Comment by gibrown** [comment]
@rcanepa more of an fyi since you had just been changing the settings code in https://github.com/Automattic/wp-calypso/pull/46163. This uses the same language for both jetpack and wp.com sites.

File: client/my-sites/site-settings/search.jsx [hunk 1]
```diff
@@ -94,151 +93,102 @@ class Search extends Component {
...
+								) }
+								disabled={ isRequestingSettings || isSavingSettings }
+							/>
+						) : (
+							<CompactFormToggle
+								checked={ !! fields.jetpack_search_enabled }
+								disabled={ isRequestingSettings || isSavingSettings }
+								onChange={ handleAutosavingToggle( 'jetpack_search_enabled' ) }
+							>
+								{ translate( 'Replace WordPress built-in search a higher performance query.' ) }
```
Comment (bluefuton) [inline client/my-sites/site-settings/search.jsx:163]: Missing "with".

**Comment by bluefuton** [comment]
"eplace WordPress built-in search with a higher performance query" seems a bit techie.

How about "Speed up WordPress built-in search with Jetpack" or similar? (Shall we ask for a copy review?)

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Everything appears to work as expected, except one exception: Trying to purchasing Jetpack Search via `/purchase-product/jetpack_search/monthly` doesn't work. Pressing `Get Search` puts the page into a loading state indefinitely.

EDIT: Full repro steps:
1. Navigate to /purchase-product/jetpack_search/monthly.
2. Enter a simple or private simple site URL.
3. Click Get Search.
4. Interface gets stuck in the loading state indefinitely.

**Review by jsnmoon — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4885619

Thank you @gibrown for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4885619).

**Pull request merged**